3rdparty/obiwarp/vec.cpp

```diff
@@ -1436,95 +1178,63 @@ VecD VecD::operator+(const VecD &A) {
         return blank;
     }
 
-    else {
-        VecD *C = new VecD(_n);
-        VecD tmp = *C;
-        tmp._to_pass_up = C; 
+else {
+    VecD *C = new VecD(_n);
+    VecD tmp = *C;
+    tmp._to_pass_up = C;
         printf("TMPENEW %d\n", tmp.shallow());
-        for (int i = 0; i < _n; ++i) {
-            tmp[i] = _dat[i] + A[i];
-        }
-        return tmp;
+    for (int i = 0; i < _n; ++i) {
+        tmp[i] = _dat[i] + A[i];
     }
+    return tmp;
+}
 }
 */
 
 double VecD::sum() {
-    double *me = (double*)(*this);
-    double sum = 0;
-    for( int n = 0; n < _n; n++) {
-        sum += me[n];
-    }
-    return sum;
+    return std::accumulate(_dat.begin(), _dat.end(), 0.0f);
```

Comment:
Yes! Absolutely correct. In fact, in case of `std::accumulate` the third argument very much decides the kind of operations performed in intermediate steps as well the resulting value. Passing `0.0f` here makes us lose precision on each step of accumulation. Nice eye! Will change this.